client/src/document/index.tsx

```diff
@@ -366,3 +382,83 @@ function LoadingError({ error }) {
     </div>
   );
 }
+
+interface FlatFlaw {
+  name: string;
+  flaws: string[];
+  count: number;
+}
+
+function ToggleDocmentFlaws({ doc }: { doc: Doc }) {
+  const { flaws }: { flaws: object } = doc;
+  const [show, toggle] = useReducer((v) => !v, false);
+
+  useEffect(() => {
+    if (window.location.hash && window.location.hash === "#show-flaws") {
+      toggle();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (show) {
+      if (!(window.location.hash && window.location.hash === "#show-flaws")) {
+        window.location.hash = "#show-flaws";
+      }
+    } else {
+      if (window.location.hash && window.location.hash === "#show-flaws") {
+        window.location.hash = "";
+      }
+    }
+  }, [show]);
```

Comment:
```suggestion
const FLAWS_HASH = "#show-flaws";
function toggleLocationFlawsHash(on: boolean) {
  window.history.pushState(
    "",
    document.title,
    window.location.href.split("#")[0] + (on ? FLAWS_HASH : "")
  );
}

function ToggleDocumentFlaws({ doc }: { doc: Doc }) {
  const { flaws } = doc;
  const [show, toggle] = useReducer((v) => !v, false);

  useEffect(() => {
    if (window.location.hash && window.location.hash === FLAWS_HASH) {
      toggle();
    }
  }, []);

  useEffect(() => {
    const hasShowHash = window.location.hash === FLAWS_HASH;
    if (show && !hasShowHash) {
      toggleLocationFlawsHash(true);
    } else if (!show && hasShowHash) {
      toggleLocationFlawsHash(false);
    }
  }, [show]);
```

Imo the hash change shouldn't take control of the scroll position, so I'd recommend changing it to use `pushState` instead.